CMakeLists.txt

```diff
@@ -1,6 +1,27 @@
-project( common_clang )
 cmake_minimum_required(VERSION 3.4.3)
 
+if(NOT DEFINED BASE_LLVM_VERSION)
+  set (BASE_LLVM_VERSION 11.0.0)
+endif(NOT DEFINED BASE_LLVM_VERSION)
+set(OPENCL_CLANG_VERSION ${BASE_LLVM_VERSION}.0)
+
+if (NOT DEFINED OPENCL_CLANG_BUILD_EXTERNAL)
+  # check if we build inside llvm or not
+  if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
+    set(OPENCL_CLANG_BUILD_EXTERNAL YES)
+  endif(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
+endif (NOT DEFINED OPENCL_CLANG_BUILD_EXTERNAL)
+
+if(OPENCL_CLANG_BUILD_EXTERNAL)
```

Comment:
Do we need to update README, users need to define this variable if they want to build opencl-clang with pre-build llvm?

Reply:
updated, thanks.